dot/peerset/peerset_test.go

```diff
@@ -11,57 +11,103 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
-func TestPeerSetBanned(t *testing.T) {
+const testSetID = 0
+
+func TestBanRejectAcceptPeer(t *testing.T) {
 	t.Parallel()
 
 	handler := newTestPeerSet(t, 25, 25, nil, nil, false)
 
 	ps := handler.peerSet
-	require.Equal(t, unknownPeer, ps.peerState.peerStatus(0, peer1))
-	ps.peerState.discover(0, peer1)
+
+	checkPeerStateSetNumIn(t, ps.peerState, testSetID, 0)
+	peer1Status := ps.peerState.peerStatus(testSetID, peer1)
+	require.Equal(t, unknownPeer, peer1Status)
+
+	ps.peerState.discover(testSetID, peer1)
 	// adding peer1 with incoming slot.
-	err := ps.peerState.tryAcceptIncoming(0, peer1)
+	err := ps.peerState.tryAcceptIncoming(testSetID, peer1)
 	require.NoError(t, err)
 
+	checkPeerStateSetNumIn(t, ps.peerState, testSetID, 1)
+	peer1Status = ps.peerState.peerStatus(testSetID, peer1)
+	require.Equal(t, connectedPeer, peer1Status)
+
 	// we ban a node by setting its reputation under the threshold.
 	rpc := newReputationChange(BannedThresholdValue-1, "")
+
 	// we need one for the message to be processed.
+	// report peer will disconnect the peer and set the `lastConnected` to time.Now
 	handler.ReportPeer(rpc, peer1)
-	time.Sleep(time.Millisecond * 100)
 
+	time.Sleep(100 * time.Millisecond)
```

Comment:
nit can we have this event-driven? Can this sleep be removed?
Maybe by mocking `time.Now` wherever it's used?